src/ccall.cpp

```diff
@@ -96,12 +165,128 @@ static Value *runtime_sym_lookup(PointerType *funcptype, const char *f_lib, cons
     builder.CreateStore(llvmf, llvmgv);
     builder.CreateBr(ccall_bb);
 
-    ctx->f->getBasicBlockList().push_back(ccall_bb);
+    f->getBasicBlockList().push_back(ccall_bb);
     builder.SetInsertPoint(ccall_bb);
     llvmf = builder.CreateLoad(llvmgv);
     return builder.CreatePointerCast(llvmf,funcptype);
 }
 
+static Value *runtime_sym_lookup(PointerType *funcptype, const char *f_lib,
+                                 const char *f_name, Function *f)
+{
+    GlobalVariable *libptrgv;
+    GlobalVariable *llvmgv;
+    bool runtime_lib = runtime_sym_gvs(f_lib, f_name, f->getParent(),
+                                       libptrgv, llvmgv);
+    libptrgv = prepare_global(libptrgv);
+    llvmgv = prepare_global(llvmgv);
+    return runtime_sym_lookup(funcptype, f_lib, f_name, f, libptrgv, llvmgv,
+                              runtime_lib);
+}
+
+// Map from distinct callee's to it's GOT entry.
```

Comment:
"callee's" is possessive, but there's no subject there. Also very unclear what "it" is referring to.